File: src/main/java/com/example/car_manager/model/Route.java
package com.example.car_manager.model;

import jakarta.persistence.*;
import java.time.LocalTime;

@Entity
@Table(name = "routes")
public class Route {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @Column(name = "start_time", nullable = false)
    private LocalTime startTime;

    @Column(name = "finish_time", nullable = false)
    private LocalTime finishTime;

    @ManyToOne
    @JoinColumn(name = "car_id", referencedColumnName = "id")
    private Car car;

    @ManyToOne
    @JoinColumn(name = "user_id", referencedColumnName = "id")
    private User user;


}

File: frontend/app/src/main/java/com/example/frontend/model/Route.kt
package com.example.frontend.model


data class Route(
    val id: Int,
    val avgSpeed: Int,
    val distanceTraveled: Int,
    val avgFuelConsumption: Double,
    val fuelUsed: Double,
    val startTime: String,
    val finishTime: String,
    val carId: Int,
    val userId: Int
)
